File: domain/summary.go
package domain

import "time"

type MonthlySummary struct {
	Year          int
	Month         time.Month
	TotalExpenses int64
	TotalIncome   int64
	Balance       int64
	EntryCount    int
}

func NewMonthlySummary(year int, month time.Month, expenses, income int64, count int) MonthlySummary {
	return MonthlySummary{
		Year:          year,
		Month:         month,
		TotalExpenses: expenses,
		TotalIncome:   income,
		Balance:       income - expenses,
		EntryCount:    count,
	}
}

File: src/domain/summary.ts
export interface MonthlySummary {
  year: number;
  month: number;
  totalExpenses: number;
  totalIncome: number;
  totalInvest: number;
  totalSaving: number;
  balance: number;
  entryCount: number;
}

export function newMonthlySummary(
  year: number,
  month: number,
  expenses: number,
  income: number,
  count: number,
  invest = 0,
  saving = 0,
): MonthlySummary {
  return {
    year,
    month,
    totalExpenses: expenses,
    totalIncome: income,
    totalInvest: invest,
    totalSaving: saving,
    balance: income - expenses - invest - saving,
    entryCount: count,
  };
}
